nav2_behavior_tree/plugins/action/remove_passed_goals_action.cpp

```diff
@@ -31,12 +31,44 @@ RemovePassedGoals::RemovePassedGoals(
   viapoint_achieved_radius_(0.5)
 {
   getInput("radius", viapoint_achieved_radius_);
-
-  getInput("global_frame", global_frame_);
-  getInput("robot_base_frame", robot_base_frame_);
+  
   tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
   auto node = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
   node->get_parameter("transform_tolerance", transform_tolerance_);
+
+  if(!getInput("global_frame", global_frame_)) {
```

Comment:
I assume these 4 blocks are essentially the same, so I'll review this one but apply to all the other 4 blocks in the other BT nodes.

This is alot of logic. This feels like we might be served by having this be an external function that these all use to abstract out the code 1 time and use `N` times (and make it easily unit testable)

Reply:
Where should this function be inserted? I could create a bt_utils in nav2_utils for example, as I didn't find a class that all bt nodes extend from.

Reply:
Rename this file to bt_utils + add it here: https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/include/nav2_behavior_tree/bt_conversions.hpp

Reply:
Renamed and added the function with some template arguments.